src/main/webapp/app/account/register/register.component.ts

```diff
@@ -1,69 +1,107 @@
-import { NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
 import { JhiLanguageService } from 'ng-jhipster';
 
 import { RegisterService } from 'app/account/register/register.service';
 import { User } from 'app/core/user/user.model';
 import { EMAIL_ALREADY_USED_TYPE, LOGIN_ALREADY_USED_TYPE } from 'app/shared/constants/error.constants';
+import { ProfileService } from 'app/shared/layouts/profiles/profile.service';
+import { FormBuilder, Validators } from '@angular/forms';
 
 @Component({
     selector: 'jhi-register',
     templateUrl: './register.component.html',
 })
 export class RegisterComponent implements OnInit, AfterViewInit {
-    confirmPassword: string;
-    doNotMatch?: string;
-    error?: string;
-    errorEmailExists?: string;
-    errorUserExists?: string;
-    registerAccount: User;
-    success: boolean;
-    modalRef: NgbModalRef;
+    @ViewChild('login', { static: false })
+    login?: ElementRef;
 
-    constructor(private languageService: JhiLanguageService, private registerService: RegisterService, private elementRef: ElementRef, private renderer: Renderer2) {}
+    doNotMatch = false;
+    error = false;
+    errorEmailExists = false;
+    errorUserExists = false;
+    success = false;
 
-    ngOnInit() {
-        this.success = false;
-        this.registerAccount = new User();
+    defaultEmailPattern = '^[a-zA-Z0-9!$&*+=?^_`{|}~.-]+@[a-zA-Z0-9-]+(?:\\.[a-zA-Z0-9-]+)*$|^[_.@A-Za-z0-9-]+$';
+
+    registerForm = this.fb.group({
+        firstName: ['', [Validators.required, Validators.minLength(2)]],
+        lastName: ['', [Validators.required, Validators.minLength(2)]],
+        login: ['', [Validators.required, Validators.minLength(4), Validators.maxLength(50), Validators.pattern(this.defaultEmailPattern)]],
+        email: ['', [Validators.required, Validators.minLength(5), Validators.maxLength(254), Validators.email]],
+        password: ['', [Validators.required, Validators.minLength(8), Validators.maxLength(50)]],
+        confirmPassword: ['', [Validators.required, Validators.minLength(8), Validators.maxLength(50)]],
+    });
+    isRegistrationEnabled = false;
+    allowedEmailPattern?: string;
+    allowedEmailPatternReadable?: string;
+
+    constructor(private languageService: JhiLanguageService, private registerService: RegisterService, private fb: FormBuilder, private profileService: ProfileService) {}
+
+    ngAfterViewInit(): void {
+        if (this.login) {
+            this.login.nativeElement.focus();
+        }
     }
 
-    ngAfterViewInit() {
-        this.renderer.selectRootElement('#login', true).focus();
+    // TEST CODE BEGIN
+    // updateAllowedEmailPattern() {
+    //     if (this.allowedEmailPattern) {
+    //         this.registerForm.get('email')!
+    //         .setValidators([Validators.required, Validators.minLength(4), Validators.maxLength(50), Validators.pattern(this.allowedEmailPattern)]);
+    //     }
+    // }
+    // TEST CODE END
```

Comment:
Is this still needed?
```suggestion
```

Reply:
It really helped me in testing the regex pattern, so this might be super useful for other administrators playing around with Artemis, therefore I would like to keep it.